Game/Assets/FoF/Scripts/Player.cs

```diff
@@ -0,0 +1,103 @@
+﻿using UnityEngine;
+using System.Collections;
+using FoF;
+
+namespace FoF
+{
+
+	public class Player : Actor
+	{
+		public float speed = 6f;
+
+		private Rigidbody rb;
+		private Animator anim;
+		private Controls controls;
+
+		private float distToGround;
+		private Transform cameraTransform;
+		private float cameraRotX = 0;
+		
+		// For jumps
+		public float jumpStrength = 50.0f;
+
+		public bool grounded
+		{
+			get
+			{
+				// FIXME: There's a better way to do this involving colliders I think
+				return Physics.Raycast(transform.position, -Vector3.up, distToGround + 0.1F);
+			}
+		}
+
+		// Unity's initalization
+		void Start ()
+		{
+			if (Application.platform != RuntimePlatform.OSXEditor && Application.platform == RuntimePlatform.OSXPlayer && UnityEngine.VR.VRDevice.isPresent)
+			{
+				UnityEngine.VR.InputTracking.Recenter();
+				UnityEngine.VR.VRSettings.showDeviceView = true;
+				UnityEngine.VR.VRSettings.loadedDevice = UnityEngine.VR.VRDeviceType.Oculus;
+			}
+
+			anim = GetComponent<Animator>();
+			rb = GetComponent<Rigidbody>();
+			distToGround = GetComponent<Collider>().bounds.extents.y;
+			cameraTransform = transform.Find("ScientistSkeleton/Hips/Spine/Spine1/Neck/Head/Camera");   // Gotta love long identifiers
+			cameraRotX = cameraTransform.localEulerAngles.x;
+			controls = new Controls(this);
+		}
+		
+		// We shouldn't have to update here. Controls should move us around with its FixedUpdate()
+		void FixedUpdate ()
+		{
+			controls.FixedUpdate ();
```

Comment:
Turns out, you can't create a new MonoActivity with new, so I ended up having to do this to get it to actually update.